contracts/masset/Masset.sol

```diff
@@ -431,7 +418,7 @@ contract Masset is
         uint256 _mAssetQuantity,
         uint256[] calldata _minOutputQuantities,
         address _recipient
-    ) external override nonReentrant whenNoRecol returns (uint256[] memory outputQuantities) {
+    ) external override nonReentrant returns (uint256[] memory outputQuantities) {
```

Comment:
Let's think better on this scenario.
if a token is depegging a user can withdraw the token, and then when the token is back the user  just gets more tokens, and it's less tokens in the pool.

Reply:
Let's add more tests to this scenario, and let's write a test of manipulation of this flow: 
1. token price drops, 2. deposit -> withdraw loop untill the depeg is realized

I suggest to start with a close range of weights to we can minimize this affect.

Reply:
@leonprou when we pause the contracts when the token price drops, users won't be allowed to deposit and swap but only withdraw proportionate amounts, so they can't arbitrage the pool. We then unpause the contract when the contract when the token gets back to peg.